### PR TITLE
feat: #572 training improvement

### PR DESCRIPTION
**Pull request by Anok666**
https://github.com/cyfronet-fid/eosc-search-service/issues/572

### CODE REVIEW

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Keywords in all catalogs 
image
When I click keywords from the training page
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
image
"TARGET_USER": [
    {
      "id": "target_user-businesses",
      "name": "Businesses",
      "description": "An organization or economic system where goods and services are exchanged for one another or for money. Businesses can be privately owned, not-for-profit or state-owned.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-funders",
      "name": "Funders",
      "description": "Individual or organization financing a part or all of a project's cost as a grant, investment, or loan.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-innovators",
      "name": "Innovators",
      "description": "The group which is the first to try new ideas, processes, goods and services. Innovators are followed by early adopters, early majority, late majority, and laggards, in that order.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-other",
      "name": "Other",
      "description": "null",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-policy_makers",
      "name": "Policy Makers",
      "description": "Individuals (usually members of the board of directors) who have the authority to set the policy framework of an organization.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-providers",
      "name": "Providers",
      "description": "A Provider is an organisation that provides different kind of solutions and/or services or other Resources to end users and other organizations. This broad term incorporates all businesses and organisations that provide products and solutions that are offered for free, on-demand, pay per use or a hybrid delivery model.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-publishers",
      "name": "Publishers",
      "description": null,
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_communities",
      "name": "Research Communities",
      "description": "Research communities provide an infrastructure through which scientists of discipline-specific scientific areas are able to advance their research goals, reaching out to other researchers.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_groups",
      "name": "Research Groups",
      "description": "A research group is a group of researchers working together on a particular issue or topic. Research groups may be composed of researchers all from the same subject/discipline or from different subjects/disciplines.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_infrastructure_managers",
      "name": "Research Infrastructure Managers",
      "description": "A RI Manager is a type of Project Coordinator who specializes in research infrastructures. They are responsible for things like managing researchers, making sure costs are on budget and serving as a liaison between reserach staff and project stakeholders.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_managers",
      "name": "Research Managers",
      "description": "Someone in an organization whose job is to manage a research initiative aiming to the development of new scientific results, products or ideas.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_networks",
      "name": "Research Networks",
      "description": "Research networks aim to stimulate interaction between researchers and promote information exchange.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_organisations",
      "name": "Research Organisations",
      "description": "A public or private legal entity (e.g. academia, business, industry, public services, etc.) representing the User.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-research_projects",
      "name": "Research Projects",
      "description": "A privately or publicly funded project on a research topic.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-researchers",
      "name": "Researchers",
      "description": "Someone who conducts scientific research, i.e., an organized and systematic investigation by using scientific methods.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-resource_managers",
      "name": "Resource Managers",
      "description": "Resource Managers are typically responsible for managing Service level agreements with customers and external Providers.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-resource_provider_managers",
      "name": "Provider Managers",
      "description": "A Provider Manager is an individual within an organisation that is responsible for the quality of the Resources provided and monitors the delivery of the Resource.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    },
    {
      "id": "target_user-students",
      "name": "Students",
      "description": "A person who is studying at a university or other place of higher education.",
      "parentId": null,
      "type": "Target user",
      "extras": {}
    }
  ],

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

this is still wrong (from tiket)

image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV


image



"TR_URL_TYPE": [
    {
      "id": "tr_url_type-ark",
      "name": "ARK",
      "description": "Archival Resource Key",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-doi",
      "name": "DOI",
      "description": "Digital Object Identifier",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-handle",
      "name": "HANDLE",
      "description": "Handle",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-not_applicable",
      "name": "NOT APPLICABLE",
      "description": null,
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-uri",
      "name": "URI",
      "description": "Uniform Resource Identifier",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-url",
      "name": "URL",
      "description": "Uniform Resource Locator",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    },
    {
      "id": "tr_url_type-urn",
      "name": "URN",
      "description": "Uniform Resource Name",
      "parentId": null,
      "type": "Training Resource url type",
      "extras": {}
    }
  ],

**Comment by Michal-Kolomanski** [comment]
image
When duration is not provided - it should also have `-` sign

File: ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html [hunk 1]
```diff
@@ -26,84 +26,99 @@
...
+                  <p class="section" i18n>Expertise Level<br><span class="section-description">{{ myTraining.level_of_expertise }}</span></p>
+                  <p class="section" i18n>Qualification<br><span class="section-description">{{ '-' }}</span></p>
+                  <p class="section" i18n>Content Resource Type<br>
+                    <ng-container *ngFor="let resource of myTraining.content_type">
+                      <span class="section-description">{{  resource }}</span><br>
+                    </ng-container>
+                  </p>
+                </div>
+                <div class="col-4">
+                  <p class="section" i18n>Duration<br><span class="section-description">{{  myTraining.duration }}</span></p>
```
Comment (Michal-Kolomanski) [inline ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html:84]: Duration can also be empty - so '-' will be useful here as well

**Review by michal-szostak — APPROVED** [review]

File: ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html [hunk 2]
```diff
@@ -26,84 +26,99 @@
...
+              <div class="column-container">
+                <div class="col-4">
+                  <p class="section" i18n>Version Date<br><span class="section-description">{{ myTraining.publication_date | transformDatePipe }}</span></p>
+                  <p class="section" i18n>License<br><span class="section-description">{{ myTraining.license }}</span></p>
+                  <p class="section" i18n>Resource URL type<br><span class="section-description">{{ myTraining?.url_type ?? '-' | filterPipe: 'url_type'  }}</span></p>
+                </div>
+                <div class="col-4">
+                  <p class="section" i18n>Authors<br>
+                    <ng-container *ngFor="let author of myTraining.author_names">
+                      <span class="section-description">{{ author}}, </span>
```
Comment (Michal-Kolomanski) [inline ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html:56]: @agpul @Marcelinna @aleksandrahojszyk 
I wonder if a comma at he end is necessary here?

Examples:
Author, 
Author, author,

Shouldnt be just?
Author
Author, author

**Review by Michal-Kolomanski — COMMENTED** [review]
Good job! Just slight remarks. 
Please consult with @Marcelinna about which fields in trainings are not obligatory and prepare "-" for them in case they are missing.

**Comment by Michal-Kolomanski** [comment]
@Anok666, after consulting with @Marcelinna the conclusion is that: 
- our front needs to be able to handle missing values for all properties. So prepare "-" accordingly. On the other hand, I think that title, description, and type always be there.

File: ui/apps/ui/src/app/dictionary/runDictionaryForInteroperabilityGuidelines.ts [hunk 3]
```diff
@@ -1,20 +1,33 @@
...
-import { interoperabilityGuidelinesTypeDictionary } from '../dictionary/interoperabilityGuidelinesTypeDictionary';
-import { interoperabilityGuidelinesDomainDictionary } from '../dictionary/interoperabilityGuidelinesDomainDictionary';
+import { interoperabilityGuidelinesStatusDictionary } from './interoperabilityGuidelinesStatusDictionary';
+import { interoperabilityGuidelinesTypeDictionary } from './interoperabilityGuidelinesTypeDictionary';
+import { interoperabilityGuidelinesDomainDictionary } from './interoperabilityGuidelinesDomainDictionary';
+import { trainingAccessDictionary } from './trainingAccessDictionary';
+import { trainingTargetGroupDictionary } from './trainingTargetGroupDictionary';
+import { trainingUrlTypeDictionary } from './trainingUrlTypeDictionary';
 
 export function runDictionaryForInteroperabilityGuidelines(
```
[the remark this answers is not in the document]
Comment (Anok666) [reply ui/apps/ui/src/app/dictionary/runDictionaryForInteroperabilityGuidelines.ts]: rename function to translateDictionaryValue

File: ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html (hunk 1 above)
Comment (Anok666) [reply ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html]: done

File: ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html (hunk 2 above)
Comment (Anok666) [reply ui/apps/ui/src/app/pages/trainings-page/training-detail-page.component.html]: done

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

Please remove "_" from target user
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
All more space
image

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**